tests/Unit/Parameters/ParameterTranslatorTest.php

```diff
@@ -0,0 +1,132 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA;
+ */
+
+namespace Oat\DbalSpanner\Tests\Unit\Parameters;
+
+use Doctrine\DBAL\Exception\InvalidArgumentException;
+use OAT\Library\DBALSpanner\Parameters\ParameterTranslator;
+use OAT\Library\DBALSpanner\Tests\_helpers\NoPrivacyTrait;
+use PHPUnit\Framework\TestCase;
+
+class ParameterTranslatorTest extends TestCase
+{
+    use NoPrivacyTrait;
+
+    /** @var ParameterTranslator */
+    protected $subject;
+
+    public function setUp(): void
+    {
+        $this->subject = new ParameterTranslator();
+    }
+
+    /**
+     * @dataProvider placeHoldersToTest
+     */
+    public function testTranslatePlaceHolders($sql, $expected, $expectedPositionalParameterCount)
+    {
+        $this->assertEquals($expected, $this->subject->translatePlaceHolders($sql));
+        $this->assertEquals($expectedPositionalParameterCount, $this->getPrivateProperty($this->subject, 'positionalParameterCount'));
+    }
+
+    public function placeHoldersToTest()
+    {
+        $expectedNamed = 'SELECT * FROM statements WHERE modelid = @model AND subject = @subject';
+        $expectedPositional = 'SELECT * FROM statements WHERE modelid = @param1 AND subject = @param2';
+
+        return [
+            ['', '', 0],
+            ['SELECT * FROM statements WHERE modelid = :model AND subject = :subject', $expectedNamed, 0],
+            ['SELECT * FROM statements WHERE modelid = @model AND subject = @subject', $expectedNamed, 0],
+            ['SELECT * FROM statements WHERE modelid = @model AND subject = :subject', $expectedNamed, 0],
+            ['SELECT * FROM statements WHERE modelid = ? AND subject = ?', $expectedPositional, 2],
+            ['SELECT * FROM statements WHERE modelid = ? AND subject = ?;', $expectedPositional . ';', 2],
+        ];
+    }
+
+    /**
+     * @dataProvider mixedParameterSyntaxesToTest
+     */
+    public function testTranslatePlaceHoldersWithMixedTypesThrowsException(string $sql)
+    {
+        // Creates a statement with a blank sql string to avoid detection on constructor.
+        $this->expectException(InvalidArgumentException::class);
+        $this->subject->translatePlaceHolders($sql);
+    }
+
+    public function mixedParameterSyntaxesToTest()
+    {
+        return [['?@'], ['?:'], ['?:@']];
+    }
+
+    /**
+     * @dataProvider namedParamsToTest
+     */
+    public function testConvertPositionalToNamedWithNamedParameters(array $expected, array $boundValues, array $params = null)
+    {
+        $this->assertEquals($expected, $this->subject->convertPositionalToNamed($boundValues, $params));
```

Comment:
I find this way of representing unit tests more readable. 
```suggestion
        $result = $this->subject->convertPositionalToNamed($boundValues, $params);
        $this->assertEquals($expected, $result);
```